File: src/ecs/entities.rs
use crate::ecs::components as C;
use crate::tileset::Tile;
use hecs::*;
use macroquad::prelude::*;

pub fn player(ecs: &mut World, pos: &Vec2) {
    ecs.spawn((
        C::Position::new(pos.x, pos.y),
        C::Display {
            tile: Tile::MAN,
            color: WHITE,
        },
        C::Player,
    ));
}

pub fn tree(ecs: &mut World, pos: &Vec2) {
    ecs.spawn((
        C::Position::new(pos.x, pos.y),
        C::Display {
            tile: Tile::TREE,
            color: GREEN,
        },
    ));
}

File: Yaldabaranth.Core/Scripts/ECS/Entities/Entities.cs
using Friflo.Engine.ECS;
using Yaldabaranth.Core.Scripts.ECS.Components;
using Microsoft.Xna.Framework;
using Yaldabaranth.Core.Scripts.ECS.Tags;

namespace Yaldabaranth.Core.Scripts.ECS.Entities;

public static class E
{
  public static void SpawnPlayer(YaldabaranthGame game, int x, int y)
  {
    var e = game.Entities.CreateEntity(
      new C.Position(game, new Vector2(x, y)),
      new C.Velocity(new Vector2(0, 0)),
      new C.Player(),
      new C.Display(Tile.Man, Color.White),
      new C.Eyes(game)
    );
    e.AddTag<T.Visible>();
    e.AddTag<T.Opaque>();
  }
  public static void SpawnTree(YaldabaranthGame game, int x, int y)
  {
    var e = game.Entities.CreateEntity(
      new C.Position(game, new Vector2(x, y)),
      new C.Velocity(new Vector2(0, 0)),
      new C.Display(T: Tile.Tree, C: Color.Yellow)
    );
    e.AddTag<T.Visible>();
    e.AddTag<T.Opaque>();
  }
}
